modules/apps/forms-and-workflow/dynamic-data-lists/dynamic-data-lists-form-web/src/main/resources/META-INF/resources/admin/js/form_builder_field_settings_sidebar.js

```diff
@@ -50,7 +50,8 @@ AUI.add(
 
 						eventHandlers = [
 							instance.after('open', instance._afterSidebarOpen),
-							instance.after('open:start', instance._afterOpenStart)
+							instance.after('open:start', instance._afterOpenStart),
+							instance.after('openChange', instance._afterOpenChange)
```

Comment:
@thiago-rocha. Here is the commit that implements close sidebar by clicking outside.